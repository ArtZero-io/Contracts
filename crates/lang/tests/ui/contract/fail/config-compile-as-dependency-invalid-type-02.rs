use ink_lang as ink;

#[ink::contract(compile_as_dependency = 42)]
mod contract {
    #[ink(storage)]
    pub struct Contract {}

    impl Contract {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
