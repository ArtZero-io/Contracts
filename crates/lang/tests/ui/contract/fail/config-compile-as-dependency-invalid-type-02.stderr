error: expected a bool literal for `compile_as_dependency` ink! configuration argument
 --> tests/ui/contract/fail/config-compile-as-dependency-invalid-type-02.rs:3:17
  |
3 | #[ink::contract(compile_as_dependency = 42)]
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
